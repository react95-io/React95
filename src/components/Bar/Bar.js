import React from 'react';
import propTypes from 'prop-types';
import styled from 'styled-components';

const StyledBar = styled.div`
  display: inline-block;
  box-sizing: border-box;
  height: ${({ size }) => size};
  width: 5px;
  border-top: 2px solid ${({ theme }) => theme.borderLightest};
  border-left: 2px solid ${({ theme }) => theme.borderLightest};
  border-bottom: 2px solid ${({ theme }) => theme.borderDark};
  border-right: 2px solid ${({ theme }) => theme.borderDark};
  background: ${({ theme }) => theme.material};
`;
// TODO: add horizontal variant
// TODO: allow user to specify number of bars (like 3 horizontal bars for drag handle)
const Bar = React.forwardRef(function Bar(props, ref) {
  const { size: sizeProp, ...otherProps } = props;
  const size = typeof sizeProp === 'number' ? `${sizeProp}px` : sizeProp;

  return <StyledBar size={size} ref={ref} {...otherProps} />;
});

Bar.defaultProps = {
  size: '100%'
};
Bar.propTypes = {
  size: propTypes.oneOf([propTypes.string, propTypes.number])
};

export default Bar;
