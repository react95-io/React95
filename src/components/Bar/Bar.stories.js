import React from 'react';

import { Bar, AppBar, Toolbar, Button } from 'react95';

export default {
  title: 'Bar',
  component: Bar,
  decorators: [
    story => (
      <div
        style={{
          padding: '5rem',
          background: 'teal'
        }}
      >
        {story()}
      </div>
    )
  ]
};

export const Default = () => (
  <AppBar>
    <Toolbar>
      <Bar size={35} />
      <Button variant='menu'>Edit</Button>
      <Button variant='menu' disabled>
        Save
      </Button>
      <Bar size={35} />
    </Toolbar>
  </AppBar>
);

Default.story = {
  name: 'default'
};
