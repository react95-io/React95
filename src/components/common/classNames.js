/**
 * Given a string, number, object, or array, extract
 * the class names configuration from it, and return
 * as a single class string.
 * @param {*} [value]
 * @return {string}
 */
export const getClassesForValue = value => {
  if (value) {
    if (typeof value === 'string' || typeof value === 'number') {
      return value.toString();
    }

    if (Array.isArray(value)) {
      return value.map(getClassesForValue).join(' ');
    }

    if (typeof value === 'object' && !Array.isArray(value)) {
      return Object.keys(value)
        .filter(key => Boolean(value[key]))
        .map(getClassesForValue)
        .join(' ');
    }
  }

  return '';
};

/**
 * Given any number of arguments of type string, number,
 * object, or a multidimensional array, extract a class names
 * configuration from it, and return as a single classes string.
 * For object values, uses keys as classes, but only if their
 * associated values are truthy.
 * @example
 * const result = classNames('alpaca', ['bear', 'cat']);
 * --> 'alpaca bear cat'
 * const result = classNames('alpaca', { bear: null, cat: true });
 * --> 'alpaca cat' // 'bear' is null, so is not included
 * const result = classNames(['alpaca', { bear: true }, ['cat']]);
 * --> 'alpaca bear cat' // multi-dim array is flattened
 * @param {...*} args
 * @return {string}
 */
const classNames = (...args) => {
  return args.reduce((classesString, arg) => {
    const classesForArg = getClassesForValue(arg);

    // leading whitespace is okay as the final string is trimmed
    return `${classesString} ${classesForArg}`.trim();
  }, '');
};

export default classNames;
