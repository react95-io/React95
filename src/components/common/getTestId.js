const getTestId = (testIdPrefix, appendText = '') =>
  testIdPrefix ? `${testIdPrefix}${appendText}`.trim() : undefined;

export default getTestId;
