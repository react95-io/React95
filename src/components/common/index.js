import { css } from "styled-components";

export const shadow = `4px 4px 10px 0 rgba(0,0,0,0.35)`;
export const insetShadow = `inset 3px 3px 10px rgba(0, 0, 0, 0.2)`;

export const createDisabledTextStyles = () => css`
  color: ${({ theme }) => theme.textDisabled};
  text-shadow: 1px 1px ${({ theme }) => theme.textDisabledShadow};
  /* filter: grayscale(100%); */
`;
export const createBoxStyles = () =>
  css`
    box-sizing: border-box;
    display: inline-block;
    background-color: ${({ theme }) => theme.material};
    color: ${({ theme }) => theme.text};
  `;
export const createBorderStyles = (invert = false) =>
  invert
    ? css`
        border-style: solid;
        border-width: 2px;
        border-left-color: ${({ theme }) => theme.borderDarkest};
        border-top-color: ${({ theme }) => theme.borderDarkest};
        border-right-color: ${({ theme }) => theme.borderLightest};
        border-bottom-color: ${({ theme }) => theme.borderLightest};
        box-shadow: ${props => props.shadow && shadow + ", "} inset 1px 1px 0px
            1px ${({ theme }) => theme.borderDark},
          inset -1px -1px 0 1px ${({ theme }) => theme.borderLight};
      `
    : css`
        border-style: solid;
        border-width: 2px;
        border-left-color: ${({ theme }) => theme.borderLightest};
        border-top-color: ${({ theme }) => theme.borderLightest};
        border-right-color: ${({ theme }) => theme.borderDarkest};
        border-bottom-color: ${({ theme }) => theme.borderDarkest};
        box-shadow: ${props => props.shadow && shadow + ", "} inset 1px 1px 0px
            1px ${({ theme }) => theme.borderLight},
          inset -1px -1px 0 1px ${({ theme }) => theme.borderDark};
      `;
