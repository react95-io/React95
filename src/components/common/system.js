// TODO - implement styled-system

export const blockSizes = {
  sm: '27px',
  md: '35px',
  lg: '43px'
};

export const fontFamily = 'sans-serif';
