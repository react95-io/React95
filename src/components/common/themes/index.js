import azureOrange from './azureOrange';
import bee from './bee';
import blackAndWhite from './blackAndWhite';
import brick from './brick';
import candy from './candy';
import coldGray from './coldGray';
import eggplant from './eggplant';
import highContrast from './highContrast';
import lilac from './lilac';
import lilacRoseDark from './lilacRoseDark';
import maple from './maple';
import marine from './marine';
import matrix from './matrix';
import millenium from './millenium';
import modernDark from './modernDark';
import molecule from './molecule';
import ninjaTurtles from './ninjaTurtles';
import olive from './olive';
import original from './original';
import pamelaAnderson from './pamelaAnderson';
import plum from './plum';
import rainyDay from './rainyDay';
import rose from './rose';
import slate from './slate';
import spruce from './spruce';
import theSixtiesUSA from './theSixtiesUSA';
import tokyoDark from './tokyoDark';
import tooSexy from './tooSexy';
import travel from './travel';
import vaporTeal from './vaporTeal';
import vermillion from './vermillion';
import violetDark from './violetDark';
import water from './water';

export default {
  azureOrange,
  bee,
  blackAndWhite,
  brick,
  candy,
  coldGray,
  eggplant,
  highContrast,
  lilac,
  lilacRoseDark,
  maple,
  marine,
  matrix,
  millenium,
  modernDark,
  molecule,
  ninjaTurtles,
  olive,
  original,
  pamelaAnderson,
  plum,
  rainyDay,
  rose,
  slate,
  spruce,
  theSixtiesUSA,
  tokyoDark,
  tooSexy,
  travel,
  vaporTeal,
  vermillion,
  violetDark,
  water
};
