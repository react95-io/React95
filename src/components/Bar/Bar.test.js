import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import PropTypes from "prop-types";
import checkPropTypes, { assertPropTypes } from "check-prop-types";

import Bar from "./Bar";
import themes from "../common/themes";
import { blockSizes } from "../common/system";

describe("Component: Bar", () => {

  const testProps = {
    className: "my-class",
    style: { color: "red" },
    size: "sm"
  }

  it("warns if props are missing or the wrong type", () => {
    expect(() => { assertPropTypes(Bar.propTypes, { ...testProps, className: 123 }, "prop", Bar.className) }).toThrow();
    expect(() => { assertPropTypes(Bar.propTypes, { ...testProps, style: "red" }, "prop", Bar.style) }).toThrow();
    expect(() => { assertPropTypes(Bar.propTypes, { ...testProps, size: true }, 'prop', Bar.size) }).toThrow();
  });

  it("renders", () => {
    const wrapper = shallow(<Bar />);
    expect(wrapper.length).toBe(1);
  });

  it("renders className", () => {
    const wrapper = shallow(<Bar className="my-class" />);
    expect(wrapper.props().className).toBe("my-class");
  });

  it("renders style object", () => {
    const wrapper = shallow(<Bar style={{ color: "blue" }} />);
    expect(wrapper.props().style.color).toBe("blue");
  });

  it("renders other props passed", () => {
    const wrapper = shallow(<Bar a={1} b={2} c={3} />);
    expect(wrapper.props().a).toBe(1);
    expect(wrapper.props().b).toBe(2);
    expect(wrapper.props().c).toBe(3);
  });

  it("renders default props", () => {
    const wrapper = shallow(<Bar />);
    expect(wrapper.props().size).toBe("md");
  });

  it("renders with prop.size passed", () => {
    const wrapper = shallow(<Bar size="lg" />);
    expect(wrapper.props().size).toBe("lg");
  });

  it("renders default component styles", () => {
    const tree = renderer.create(<Bar theme={themes.default} />).toJSON();
    expect(tree).toHaveStyleRule("display", "inline-block");
    expect(tree).toHaveStyleRule("box-sizing", "border-box");
    expect(tree).toHaveStyleRule("height", blockSizes["md"]);
    expect(tree).toHaveStyleRule("width", "5px");
    expect(tree).toHaveStyleRule("border-top", `2px solid ${themes.default.borderLightest}`);
    expect(tree).toHaveStyleRule("border-left", `2px solid ${themes.default.borderLightest}`);
    expect(tree).toHaveStyleRule("border-bottom", `2px solid ${themes.default.borderDark}`);
    expect(tree).toHaveStyleRule("border-right", `2px solid ${themes.default.borderDark}`);
    expect(tree).toHaveStyleRule("background", themes.default.material);
  });

  it("render a matching snapshot", () => {
    const component = renderer.create(
      <Bar
        size="sm"
        className="sm-gd-12"
        style={{ color: "red" }}
      />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot();
  })
})
