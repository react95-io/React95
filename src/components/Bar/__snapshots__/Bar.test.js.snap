// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Bar render a matching snapshot 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 27px;
  width: 5px;
  border-top: 2px solid;
  border-left: 2px solid;
  border-bottom: 2px solid;
  border-right: 2px solid;
}

<div
  className="sm-gd-12 c0"
  size="sm"
  style={
    Object {
      "color": "red",
    }
  }
/>
`;
